src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1302,317 +1294,320 @@ private void IfDiagnostic(CompilationAnalysisContext context, StatementSyntax st
             private bool CheckDiagnosticCreation(CompilationAnalysisContext context, SyntaxToken statementIdentifierToken, SyntaxToken keywordIdentifierToken, List<string> ruleNames, SyntaxList<StatementSyntax> statements, ParameterSyntax contextParameter)
             {
                 int statementCount = statements.Count;
-
-                SyntaxToken openParenToken = OpenParenAnalysis(statementIdentifierToken, statements);
-                if (openParenToken.Text == "")
+                if (statementCount < 9)
                 {
-                    IfDiagnostic(context, statements[3], OpenParenIncorrectRule, statementIdentifierToken.Text);
-                    return false;
-                }
-
-                if (statementCount > 4)
-                {
-                    SyntaxToken startToken = StartAnalysis(keywordIdentifierToken, statements);
-                    if (startToken.Text == "")
+                    List<string> diagnosticInfo = DiagnosticCreationCheck(ruleNames, statements[statementCount - 1]);
+                    if (diagnosticInfo.Count == 0 || diagnosticInfo[0] == "" || diagnosticInfo[1] == "")
                     {
-                        IfDiagnostic(context, statements[4], StartSpanIncorrectRule, keywordIdentifierToken.Text);
+                        IfDiagnostic(context, statements[statementCount - 1], DiagnosticIncorrectRule, ruleNames[0]);
```

Comment:
@jepetty Yes, that is my point. if the statement is incorrect in that it shouldn't even be an if-statement, it doesn't make sense to just squiggle the condition. That makes it seem like only the condition is wrong. In the specific case where the statement is wrong because it shouldn't even be an if-statement, it makes sense to squiggle everything, including the block.